File: hospitalManagement/Medicines.cs
using System;
using System.Collections.Generic;
using System.Collections.Specialized;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace hospitalManagement
{
    internal class Medicines : IBasicActivities<Medicine>
    {
        //Field
        private List<Medicine> medicineList;
        private long count;

        // Infor field
        //  Management field
        // Dynamic 
        // Properties
        public long Count { get => count; set => count = value; }
        public List<Medicine> MedicineList { get => medicineList; set => medicineList = value; }
        // Constructors
        public Medicines()
        {
            medicineList = new List<Medicine>();
            count = 0;
        }
        public Medicines(List<Medicine> medicineList, long count)
        {
            this.MedicineList = medicineList;
            this.Count = count;
        }


        // Destructors
        // Methods
        // in, output
        // General method
        // Other method
        public void AddItem()
        {
            Console.WriteLine("Add medicine");

            Medicine medicine = new Medicine();
            medicine.Input();
            medicineList.Add(medicine);
            this.Count++;
            Console.WriteLine("Done!");

        }
        public void AddItem(Medicine value)
        {
            Console.WriteLine("Add medicine");

            medicineList.Add(value);
            this.Count++;
            Console.WriteLine("Done!");

        }
        public void ShowInformation()
        {
            Console.WriteLine("Show information of all medicines");

            medicineList.ForEach(value => { value.Output(); Console.WriteLine(); });
            Console.WriteLine("Done!");

        }

        public Medicine UpdateItem(string id)
        {
            Console.WriteLine("Update the medicine");

            Medicine res = null;
            medicineList.ForEach(value =>
            {
                if (value.Id == id)
                {
                    value.Input();
                    res = value;
                }
            });
            if (res == null)
            {
                Console.WriteLine($"Not found the medicine with id:{id}");

            }
            else
            {
                Console.WriteLine("Done!");

            }
            return res;
        }

        public bool RemoveItem(string id)
        {
            Console.WriteLine("Remove the medicine");

            bool res = false;
            medicineList.ForEach(value =>
            {
                if (value.Id == id)
                {
                    medicineList.Remove(value);
                    res = true;
                    this.Count--;
                }
            });
            if (res == false)
            {
                Console.WriteLine($"Not found medicine with id: {id}");

            }
            else
            {
                Console.WriteLine("Done!");

            }
            return res;
        }

        public Medicine FindItem(string id)
        {
            Medicine res = null;
            medicineList.ForEach(value =>
            {
                if (value.Id == id)
                {
                    res = value;
                }
            });
            if (res == null)
            {
                Console.WriteLine($"Not found medicine with id: {id}");

            }
            else
            {
                Console.WriteLine("Done!");

            }
            return res;
        }

        public bool Clear()
        {
            Console.WriteLine("Clear all medicines");

            medicineList.Clear();
            if (medicineList.Count > 0)
            {
                Console.WriteLine("Failure");
                return false;
            }
            this.Count = 0;
            Console.WriteLine("Done!");
            return true;
        }



        // Overriding
    }
}

File: src/hospitalManagement/Medicines.java
package hospitalManagement;

import java.util.ArrayList;

public class Medicines implements IBasicActivities<Medicine> {

	 //Field
    private ArrayList<Medicine> medicineList;
	private long count;

    // Infor field
    //  Management field
    // Dynamic 
    // Properties
	public ArrayList<Medicine> getMedicineList() {
		return medicineList;
	}
	public void setMedicineList(ArrayList<Medicine> medicineList) {
		this.medicineList = medicineList;
	}
	public long getCount() {
		return count;
	}
	public void setCount(long count) {
		this.count = count;
	}
    // Constructors
    public Medicines()
    {
        medicineList = new ArrayList<Medicine>();
        count = 0;
    }
    public Medicines(ArrayList<Medicine> medicineList, long count)
    {
        this.medicineList = medicineList;
        this.count = count;

    }
    // Destructors
    // Methods
    // in, output
    // General method
    // Other method
    public void AddItem()
    {
        System.out.println("Add medicine:");
        Medicine medicine = new Medicine();
        medicine.Input();
        medicineList.add(medicine);
        this.count++;
        System.out.println("Done!");

    }
    public void AddItem(Medicine value)
    {
        System.out.println("Add medicine");
        medicineList.add(value);
        this.count++;
        System.out.println("Done!");

    }
    public void ShowInformation()
    {
        System.out.println("\tShow information of all medicines");
        medicineList.forEach(value -> { value.Output(); System.out.println(); });
        System.out.println("Done!");
    }

    public Medicine UpdateItem(String id)
    {
        System.out.println("Update the medicine");

        Medicine res = null;
        for (Medicine i : medicineList) {
			if( i.getId() == id)
			{
				i.Input();
				res = i;
			}
		}
        if (res == null)
        {
            System.out.format("Not found the medicine with id: %s", id);
        }
        else
        {
            System.out.println("Done!");

        }
        return res;
    }

    public boolean RemoveItem(String id)
    {
        System.out.println("Remove the medicine");

        boolean res = false;
        for (Medicine i : medicineList) {
			if( i.getId() == id)
			{
				medicineList.remove(i);
				this.count--;
			}
		}
        if (res == false)
        {
            System.out.format("Not found medicine with id: ", id);

        }
        else
        {
            System.out.println("Done!");
        }
        return res;
    }

    public Medicine FindItem(String id)
    {
        System.out.println("Find the medicine");
        Medicine res = null;
        for (Medicine i : medicineList) {
			if( i.getId() == id)
			{
				res = i;
			}
		}
        if (res == null)
        {
            System.out.format("Not found medicine with id: ", id);

        }
        else
        {
            System.out.println("Done!");

        }
        return res;
    }

    public boolean Clear()
    {
        System.out.println("Clear all medicines");

        medicineList.clear();
        return true;
    }

    // Overriding

}
